ring_buffer/ring_buffer.py

```diff
@@ -8,13 +8,33 @@ def __init__(self, capacity):
         self.storage = DoublyLinkedList()
 
     def append(self, item):
-        pass
+        if not self.storage.head and not self.storage.tail:
+            self.storage.add_to_head(item)
+            self.current = self.storage.head
+        elif self.storage.length == self.capacity and self.current.next is not None:
+            next_oldest = self.current.next
+            self.current.value = item
+            self.current = next_oldest
+        elif self.storage.length == self.capacity and self.current.next is None:
```

Comment:
So, reset it at this point; I like it! Allen and I had a way more convoluted solution; I'm glad the rest of the team came up with a much cleaner solution. :D